### ISSUE TITLE
Multi environment support

### ISSUE THREAD

**Issue by wbreza** [Azure/azure-dev#383]
Update CI/CD to support multiple environments and establish baseline for application environment promotion.

Dev > Test > Staging > Production

What needs to happen in between?
- Integration tests?
- Manual approval gates?
- Canary or blue/green rollout?

**Comment by rajeshkamal5050** [comment]
@ellismg @savannahostrowski 

The new environment proposal does not support promoting applications through various environment stages (test, dev, prod). Currently there isn't any requirement to support that in azd by mapping it to how target hosts support for the same ex: Azure App Service Slots.

Keeping it in backlog. Please adjust if you think otherwise.

**Comment by pamelafox** [comment]
To clarify: is there any way to deploy to a slot right now in App Service? I've been asked by the community a few times about blue/green deployment, and am looking into whether that can be an AZD demo or whether AZ CLI still needed. 
This article describes what I'm looking for: https://www.willvelida.com/posts/blue-green-deployments-app-service-containers

**Comment by savannahostrowski** [comment]
AFAIK we don't support deployment slots at this time (@rajeshkamal5050 correct me if I'm wrong and this has slipped by me).

We do want to enable this scenario though. We don't currently have it on the roadmap as we are tackling a lot of other features/fixes but I will put this in the Gallium milestone for investigation next semester.

**Comment by rajeshkamal5050** [comment]
@setuc could you share some concrete use-cases which you had seen come up on such customer engagements?

**Comment by jeannotdamoiseaux** [comment]
I'd like to advocate for adding deployment slot support to azd, particularly for App Service deployments. This is crucial for enabling proper multi-environment management and promotion between environments.

Deployment slots are a key feature of App Service that enable blue-green deployments, A/B testing, and staging environments. Adding a `--slot` parameter to `azd deploy` would allow deploying to specific slots, enabling workflows like deploying to staging for testing before swapping to production.

**Comment by kristenwomack** [comment]
> To clarify: is there any way to deploy to a slot right now in App Service? I've been asked by the community a few times about blue/green deployment, and am looking into whether that can be an AZD demo or whether AZ CLI still needed. This article describes what I'm looking for: https://www.willvelida.com/posts/blue-green-deployments-app-service-containers

@puicchan once the blog post is ready, we need to update this issue

**Comment by puicchan** [comment]
@kristenwomack deploy to a slot for App Service is diff from the blue-green blog which is for ACA. Will look into this later.

**Comment by rajeshkamal5050** [comment]
Closing as duplicate of #2373, which tracks the same multi-environment workflow with more detail and active discussion.

If you have a specific scenario not covered by #2373, feel free to reopen or comment there.
